scipy/stats/qmc.py

```diff
@@ -0,0 +1,648 @@
+"""Quasi-Monte Carlo methods.
+
+Define function to generate sample of points in the unit hypercube.
+
+"""
+
+from __future__ import division
+
+import copy
+import numpy as np
+from scipy.optimize import brute
+from scipy._lib._util import check_random_state
+
+try:
+    from scipy.optimize import basinhopping
+
+    have_basinhopping = True
+except ImportError:
+    have_basinhopping = False
+
+
+def discrepancy(sample, bounds=None, iterative=False):
+    """Discrepancy.
+
+    Compute the centered discrepancy on a given sample.
+    It is a measure of the uniformity of the points in the parameter space.
+    The lower the value is, the better the coverage of the parameter space is.
+
+    Parameters
+    ----------
+    sample : array_like (n_samples, k_vars)
+        The sample to compute the discrepancy from.
+    bounds : tuple or array_like ([min, k_vars], [max, k_vars])
+        Desired range of transformed data. The transformation apply the bounds
+        on the sample and not the theoretical space, unit cube. Thus min and
+        max values of the sample will coincide with the bounds.
+    iterative : bool
```

Comment:
The purpose of this is unclear, a more extensive explanation would be good.

Reply:
Will add it. The discrepancy measure the uniformity of a design. The lowest the measure, the better the design in terms of randomness and uniformity.

Reply:
I changed this a bit. I can add more if needed.